lib/src/date_only_impl.dart

```diff
@@ -0,0 +1,53 @@
+part of '../kiota_abstractions.dart';
+
+/// Implements the [DateOnly] interface.
+class DateOnlyImpl implements DateOnly {
```

Comment:
maybe this is something specific to Dart I'm not aware of, but do you expect people to "know about this implementation"? 
Why not having only a class instead of class + interface?

Reply:
I thought using an interface would make the code more testable. Also the implementation shouldn't necessarily define the public interface.

You are right that this class is not very... discoverable. I'd move the factory methods to the "interface", which then return the `DateTimeImpl`. This way, the usage looks like

```dart
DateOnly.fromDateTime(DateTime.now());
```